### ISSUE TITLE
Menu items are not in italic, sections is.

### ISSUE THREAD

**Issue by kierun** [mkdocs/mkdocs#2696]
### Description

![screenshot](https://user-images.githubusercontent.com/3079575/144606956-0f985fc0-30a1-4d64-b91e-46bcb577b19a.png)

The menu showing the different sections does not support italic. However, the section does.

Previously report in [squidfunk/mkdocs-material #3293](https://github.com/squidfunk/mkdocs-material/issues/3293) but it appears to be an upstream issue.

### Expected behaviour

The menu should be in italic as well.

### Actual behaviour

The menu is not in italic.

### Steps to reproduce

1. Create a markdown file with an italic in the title: `# ook _eek_ ook`
2. Run `mkdocs serve`
3. Look at the page, especially at the menu.

### Package versions

```bash
√ ; python --version
Python 3.10.0
√ ; mkdocs --version
mkdocs, version 1.2.3 from /home/usr/.pyenv/versions/3.10.0/lib/python3.10/site-packages/mkdocs (Python 3.10)
√ ; pip show mkdocs-material | grep -E "^Version"
Version: 8.0.2
√ ;  pip show mkdocs
Name: mkdocs
Version: 1.2.3
Summary: Project documentation with Markdown.
Home-page: https://www.mkdocs.org
Author: Tom Christie
Author-email: tom@tomchristie.com
License: BSD
Location: /home/usr/.pyenv/versions/3.10.0/lib/python3.10/site-packages
Requires: click, ghp-import, importlib-metadata, Jinja2, Markdown, mergedeep, packaging, PyYAML, pyyaml-env-tag, watchdog
Required-by: mkdocs-material
```

### Configuration

```yaml
site_name: My Top Secret stuff™
theme:
  name: material
  font:
    text: "PT Sans Narrow"
  favicon: assets/favicon.png
  palette:
    scheme: slate
    primary: teal
  logo: assets/logo.png
  features:
    - navigation.instant
markdown_extensions:
  - admonition
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight:
      use_pygments: true
      auto_title: true
      linenums: true
      linenums_style: pymdownx-inline
  - pymdownx.superfences
```

### System information

```bash
√ ; uname -a
Linux nightwatch.neverness.org 5.15.6-100.fc34.x86_64 #1 SMP Wed Dec 1 13:41:51 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
√ ; cat /etc/redhat-release
───────┬────────────────────────────────────────────────────────────────────────
       │ File: /etc/redhat-release
───────┼────────────────────────────────────────────────────────────────────────
   1   │ Fedora release 34 (Thirty Four)
───────┴────────────────────────────────────────────────────────────────────────
√ ; firefox --versino
√ ; firefox --version
Mozilla Firefox 94.0
```

**Comment by mondeja** [comment]
You can try with raw HTML tags, in this case would be `<em></em>`. For example, the next example:

```markdown
# Foo <em>bar</em> baz

Some text about.
```

...renders with the default theme as:

![image](https://user-images.githubusercontent.com/23049315/146640443-a93fa77c-3984-4c8a-87bc-40714988cb19.png)

But strange things happen with material theme:

![image](https://user-images.githubusercontent.com/23049315/146640502-cf99bbee-2dbc-414a-b3a2-35f667877719.png)


Anyways, this is not supported by Mkdocs, so it seems like a hack. Note that the toc at the left in Mkdocs theme is not formatted, because Markdown and HTML markups are removed with [Python-Markdown's `toc` extension](https://python-markdown.github.io/extensions/toc/), see https://github.com/Python-Markdown/markdown/issues/935.

**Comment by AndydeCleyre** [comment]
Same problem here with backticks for monospaced words within the title, which look pretty noisy and not mono in the TOC.

**Comment by AndydeCleyre** [comment]
I now notice that a second level section title (`##`) will also not render as mono or italics in the nav sidebar, but *will* be stripped of the corresponding asterisks and backticks.

**Comment by pawamoy** [comment]
Should be "fixed" by https://github.com/mkdocs/mkdocs/pull/3578 (coming in v1.6), which will strip any HTML tags from the titles displayed in the ToC and navigation. If we *want* to allow HTML in ToC/navigation titles, we should open a new issue to discuss it. Based on my exchanges with @waylan, for the titles in the ToC at least, MkDocs should expose the `data-toc-label` HTML attribute (`label` Python attribute?) to themes so that they can use it when rendering the ToC. Not sure about the navigation.
